### ISSUE TITLE
Starship + undistract-me results in unusable shell

### ISSUE THREAD

**Issue by l0b0** [NixOS/nixpkgs#257720]
### Describe the bug

See screenshot below - basically, I can still open a shell and type in commands, but they don't actually run at all.

### Steps To Reproduce
1. Set `programs.bash.undistractMe.enable = true;` and `programs.starship.enable = true;`
2. Run `sudo nixos-rebuild switch`
3. Start a new Console

### Expected behavior

These two features should be able to co-exist.

### Screenshots

As you can see, the prompt is no longer usable at all:

![image](https://github.com/NixOS/nixpkgs/assets/168301/aee56e35-7e80-45a2-91f8-55d162fe4515)

### Additional information

Worked around this by disabling undistract-me.

### Notify maintainers

@bbigras @danth @davidtwco @Br1ght0ne @Frostman @marsam @kira-bruneau 

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.55, NixOS, 23.05 (Stoat), 23.05.3759.261abe8a44a7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - channels(root): `"nixos-23.05"`
 - channels(username): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by Artturin** [comment]
Not related to the issue but starship seems to have some kind of undistract-me builtin https://github.com/starship/starship/blob/43b2d42cd526e34c5f0290e7409fbd6d3a54e908/src/modules/cmd_duration.rs#L65

**Comment by l0b0** [comment]
> starship seems to have some kind of undistract-me builtin

Is this the [Command Duration in the config](https://starship.rs/config/#command-duration)? I can't find any documentation about it actually doing desktop notifications.

**Comment by danth** [comment]
Perhaps the runtime counters are interfering with each other.

**Comment by 1sixth** [comment]
> I can't find any documentation about it actually doing desktop notifications.

Set `show_notifications = true` and it will show notifications for any command that took 45s or longer. I just tried and it works. `undistract-me` hasn't been updated for 3 years, if you're already using starship I see no reason to use it.

BTW, I just found [This PR](https://github.com/starship/starship/pull/5438) which mentions the same `STARSHIP_PREEXEC_READY: unbound variable` error.
